### ISSUE TITLE
Support some scenario-specific keys in [campaign]

### ISSUE THREAD

**Issue by CelticMinstrel** [wesnoth/wesnoth#7392]
It would be nice to support some scenario-specific keys such as `experience_modifier`, `do_healing`, `income`, `village_gold`, `village_support` in the `[campaign]` tag. This would mean they won't need to be specified in every scenario or side.

Inspired by #7391.

**Comment by sevu** [comment]
> … or side

Related, when placed in [multiplayer], these keys set the values for all sides. Maybe the also work in [scenario], or it could be added.
In addition, they set the map default value for these settings, so that it can be seen in the MP settings prior to starting the game.

`mp_village_gold`
`mp_village_support`
`mp_fog`

**Comment by CelticMinstrel** [comment]
Yeah, the ones on this list that belong in `[side]` should probably also be supported in `[scenario]` / `[multiplayer]`.

**Comment by stevecotton** [comment]
If the values in the `[campaign]` change, would loading a start-of-scenario save pick up the new values?

**Comment by CelticMinstrel** [comment]
That's a good question. I think the preferred answer would be "yes". I'm not sure if that would require extra work in the implementation.

**Comment by Hejnewar** [comment]
If the new values are picked up upon difficulty change, what will happen to the units that now have full xp upon recalling / comming into existence? Just level up?

**Comment by CelticMinstrel** [comment]
That's another excellent question. I suspect they'd level up as soon as they receive 1 XP but that's just a guess. You can try it right now though. After all, that issue is not dependent on this and will already work in the game by setting a difficulty-dependent experience modifier in the scenario itself.
